components/Templates/includes/auto-template/Pods_Templates_Auto_Template_Front_End.php

```diff
@@ -335,49 +297,29 @@ public function front( $content ) {
 
 			// build Pods object for current item
 			global $post;
-			$pod_name = $current_post_type;
-			$pod_item = $post->ID;
-			if ( in_the_loop() ) {
-				$pod_name = $post->post_type;
-			} else {
-				// Outside the loop in a taxonomy, we want the term
-				if ( is_tax() ) {
-					$obj      = get_queried_object();
-					$pod_name = $obj->slug;
-					$pod_item = $obj->term_id;
-				}
-			}
-
-			$pod_name_and_item = array( $pod_name, $pod_item );
-			/**
-			 * Change which pod and item to run the template against. The
-			 * default pod is the the post type of the post about to be
-			 * displayed, the default item is the post about to be displayed,
-			 * except outside the loop in a taxonomy archive, in which case it
-			 * is the term the archive is for.
-			 *
-			 * @since 2.7.16
-			 *
-			 * @param string $pod_name_and_item An array of the name of the pod to run the template against and the item (ID or slug) of the item in that pod to use.
-			 * @param string $template_source   The name of the pod from which the template was selected.
-			 * @param Post   $post              The Post object that is about to be displayed.
-			 */
-			$pod_name_and_item = apply_filters( 'pods_auto_template_pod_name_and_item', $pod_name_and_item, $current_post_type, $post );
-			$pods              = pods( $pod_name_and_item[0], $pod_name_and_item[1] );
-
-			// Heuristically decide if this is single or archive
-			$s_or_a        = 'archive';
-			$s_or_a_filter = 'archive_filter';
-			$s_or_a_append = 'archive_append';
-			if ( ! in_the_loop() || is_singular() ) {
-				$s_or_a        = 'single';
-				$s_or_a_filter = 'single_filter';
-				$s_or_a_append = 'single_append';
-			}
+			$pods = pods( $current_post_type, $post->ID );
 
-			if ( $this_pod[ $s_or_a ] && current_filter() == $this_pod[ $s_or_a_filter ] ) {
+			if ( $this_pod['single'] && is_singular( $current_post_type ) ) {
+				// load the template
+				$content = $this->load_template( $this_pod['single'], $content, $pods, $this_pod['single_append'] );
+			} //end if
+			// check if we are on an archive of the post type
+			elseif ( $this_pod['archive'] && is_post_type_archive( $current_post_type ) ) {
 				// load the template
-				$content = $this->load_template( $this_pod[ $s_or_a ], $content, $pods, $this_pod[ $s_or_a_append ] );
+				$content = $this->load_template( $this_pod['archive'], $content, $pods, $this_pod['archive_append'] );
+
+			} elseif ( is_home() && $this_pod['archive'] && $current_post_type === 'post' ) {
+				// if pfat_archive was set and we're in the blog index, try to append template
+				// append the template
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 9b75e3b45c8444cb273545c08497cefc</sub>